functional_tests/docker-compose-dev-app.yml

```diff
@@ -0,0 +1,38 @@
+services:
```

Comment:
Naming them that way around here feels odd to me, as the dockerized app instance is specifically only required for manually developing against, not for running the tests. Not a huge sticking point, but I prefer keeping the name indicative, since the pattern of use here is different, I think.